doc/release/0.19.0-notes.rst

```diff
@@ -65,6 +65,10 @@ repeatable minimizations.
 The keyword `sigma` in `scipy.optimize.curve_fit` was overloaded to also accept
 the covariance matrix of errors in the data.
 
+The method `trust-iterative` was added to the function `scipy.optimize.minimize`.
```

Comment:
I agree with you: the term interactive is confusing! I am not completely satisfied with the term "exact" but I think it is better than interactive. I will change to "trust-region-exact" for now, if one of us come up with something better we may reconsider it.